src/Data/PreciseDateTime.purs

```diff
@@ -42,32 +42,57 @@ instance boundedPreciseDateTime :: Bounded PreciseDateTime where
 instance showPreciseDateTime :: Show PreciseDateTime where
   show (PreciseDateTime dateTime ns) = "PreciseDateTime (" <> show dateTime <> ") " <> show ns
 
-nanoStringPadding = "000000000" :: String
+milliStringPadding = "000" :: String
+nanoStringPadding = "000000" :: String
+subsecondStringPadding = "000000000" :: String
 
-padNanoString :: (String -> String -> String) -> String -> String
-padNanoString fn string =
-  let padding = drop (length string) nanoStringPadding
+padString :: String -> (String -> String -> String) -> String -> String
+padString p fn string =
+  let padding = drop (length string) p
   in fn string padding
 
+padMilliString :: (String -> String -> String) -> String -> String
+padMilliString = padString milliStringPadding
+
+padNanoString :: (String -> String -> String) -> String -> String
+padNanoString = padString nanoStringPadding
+
+padSubsecondString :: (String -> String -> String) -> String -> String
+padSubsecondString = padString subsecondStringPadding
+
+leftPadMilliString :: String -> String
+leftPadMilliString = padMilliString (flip append)
+
+rightPadMilliString :: String -> String
+rightPadMilliString = padMilliString append
+
 leftPadNanoString :: String -> String
 leftPadNanoString = padNanoString (flip append)
 
 rightPadNanoString :: String -> String
 rightPadNanoString = padNanoString append
 
-parseSubseconds :: RFC3339String -> Maybe Int
+leftPadSubsecondString :: String -> String
+leftPadSubsecondString = padSubsecondString (flip append)
+
+rightPadSubsecondString :: String -> String
+rightPadSubsecondString = padSubsecondString append
+
+parseSubseconds :: RFC3339String -> Maybe String
 parseSubseconds (RFC3339String s) = do
   let parts = split (Pattern ".") s
   afterDot <- parts !! 1
   let digits = takeWhile isDigit afterDot
-  Int.fromString <<< rightPadNanoString $ digits
+  pure $ rightPadSubsecondString $ take 9 digits
```

Comment:
@natefaubion I think this might address https://github.com/awakesecurity/purescript-precise-datetime/issues/8. There's an updated test case for it, which I'll point out.